### PR TITLE
fix: put mock restore at the end

### PR DESCRIPTION
**Pull request by fengmk2**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

**Comment by fengmk2** [comment]
细节bug 真多。

File: bootstrap.js [hunk 1]
```diff
@@ -9,8 +9,9 @@ const app = mock.app(options);
 
 before(() => app.ready());
 afterEach(() => {
-  mock.restore();
-  return app.backgroundTasksFinished();
+  // restore should be the last one
+  return app.backgroundTasksFinished()
+    .then(() => mock.restore());
```
Comment (fengmk2) [inline bootstrap.js]: background task 也使用到了 mock 特性，所以必须等它们运行完才能 restore

**Comment by atian25** [comment]
说起来，cluster 那个 mockService 的 proxy 貌似在多个 worker 下也有问题。

File: bootstrap.js (hunk 1 above)
Comment (atian25) [reply bootstrap.js]: 在 catch 里面也要 restore

Comment (popomore) [reply bootstrap.js]: 用 try finally?

Comment (atian25) [reply bootstrap.js]:
```js
app.backgroundTasksFinished()
    .then(() => mock.restore())
    .catch(() => mock.restore())
```

就好了吧，egg-mock 还要支持 6 的

**Comment by fengmk2** [comment]
3.19.2

**Pull request merged**